### PR TITLE
[enhancement](nereids) improve lots of values in `insert into values` statement

### PR DESCRIPTION
**Pull request by 924060929**
## Proposed changes

improve lots of values in `insert into values` statement by bypass NereidsPlanner

the main logic is
1. `InsertUtils.normalizePlan` use `FoldConstantRuleOnFE` to reduce the expression, e.g. `values(date(now())`
2. `FastInsertIntoValuesPlanner` skip most of rules to analyze and rewrite `LogicalInlineTable` to `LogicalUnion` or `LogicalOneRowRelation`
3. fast parse date time string without date format
4. getHintMap and normal lexer share the same tokens
5. `set enable_fast_analyze_into_values=false` can force to execute all optimize rules, when we meet some bugs in `FastInsertIntoValuesPlanner`

test: insert 1000 rows with 1000 columns, the columns contains int, bigint, decimal(26,7), date, datetime, varchar(10 chinese chars)
|FastInsertIntoValuesPlanner|NereidsPlanner(enable_fast_analyze_into_values=false)| Legacy optimizer in 2.1.6 | Nereids planner in 2.1.6|
|--|--|--|--|
|16s(bottleneck is antlr's lexer)|32s|16s|80s|

If you use FastInsertIntoValuesPlanner with group commit in a transaction, the time can reduce to 12s.

TODO: build a custom lexer. in my hand write lexer test, FastInsertIntoValuesPlanner without group commit can reduce 16s to 12s, but it will take more effort: RegularExpression -> NFA -> DFA -> minimal DFA -> Lexer codegen

### CODE REVIEW

**Comment by 924060929** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

**Review by starocean999 — DISMISSED** [review]

**Comment by 924060929** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/insert/InsertIntoValuesAnalyzer.java [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+            )
+    );
+
+    public static final List<RewriteJob> BATCH_INSERT_JOBS = jobs(
+            bottomUp(
+                    new InlineTableToUnionOrOneRowRelation(),
+                    new BindSink(),
+                    new MergeProjects(),
+                    new PushProjectIntoUnion(),
+                    new PushProjectIntoOneRowRelation(),
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/insert/InsertIntoValuesAnalyzer.java:73]: these rules is expensive? why inser jobs do not contain them?

File: be/src/http/http_channel.cpp [hunk 2]
```diff
@@ -123,7 +123,8 @@ void HttpChannel::send_files(HttpRequest* request, const std::string& root_dir,
         VLOG_DEBUG << "http channel send file " << file_path << ", size: " << file_size;
 
         evbuffer_add_printf(evb.get(), "File-Name: %s\r\n", file.c_str());
-        evbuffer_add_printf(evb.get(), "Content-Length: %ld\r\n", file_size);
+        evbuffer_add_printf(evb.get(), "Content-Length: %" PRIi64 "\r\n", file_size);
```
[the remark this answers is not in the document]
Comment (924060929) [reply be/src/http/http_channel.cpp:126]: the "i64" in the "PRIi64" is the explain

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/insert/InsertIntoValuesAnalyzer.java (hunk 1 above)
Comment (924060929) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/insert/InsertIntoValuesAnalyzer.java:73]: the BatchInsert need send StringLiteral to backend, and only support `alias(literal as xx)` or `alias(cast(literal as xx))`, but not support `alias(cast(slotRef as xx))` we should push down the cast into Union all OneRowRelation.

**Comment by 924060929** [comment]
run buildall

**Review by starocean999 — APPROVED** [review]

**Review by morrySnow — APPROVED** [review]

**Pull request merged**